md-block.js

```diff
@@ -86,6 +108,10 @@ export class MarkdownElement extends HTMLElement {
 
 		marked.use({renderer: this.renderer});
 
+		if( typeof extension.extend === "function" ) {
+			marked = extension.extend(marked, this.renderer);
```

Comment:
Can't `marked` just be an export?

Reply:
we can export marked, but im not sure if it'll be the same instance, this sort of ensures the same `marked` object is being used

Reply:
It will be the same instance. Exports don't do any cloning or anything like that. The problem is that it's assigned asynchronously, so it gets tricky when to manipulate it.

I don't mind doing it like this, but this `extension.extend()` method is very ad hoc. What happens when we need extensibility in other places too? We invent different syntax?

One pattern I've used in some of my other libraries is hooks, you'd create a `Hooks` object ([reference implementation](https://github.com/LeaVerou/bliss/blob/v2/src/Hooks.js)), then you'd use

```js
let env = {marked, renderer: this.renderer, context: this};
hooks.run("render-start", env);
```

here, and 

```js
hooks.add("render-start", function({marked, renderer}) {
	/* do stuff */
});
```

in your code. `"render-start"` here is just a name for the hook, it can be anything.

You can see how hooks are used in [Prism](https://github.com/PrismJS/prism) which is probably my first popular project that used them. It's not ESM, but the only difference would be that you'd just export the hooks object.

Does that make sense?

Reply:
> but this extension.extend() method is very ad hoc

agree. 

For my case (using it in Stencil), there's already [lifecycle hooks](https://stenciljs.com/docs/component-lifecycle#component-lifecycle-methods) available, although not directly addressing the issue, they were able to put the calls in order, so I didn't give too much thought there.

I think introducing hooks is a great idea, but just might be a little overkill for the 99% of the use cases where people are happy with the default config?

Reply:
The hook implementation I liked to is like 25 loc, so not sure that's overkill for anything.

Reply:
Thx, I'll give it a go and update this PR